examples/infra/modDA.rs

```diff
@@ -0,0 +1,605 @@
+        >,
+        Storage = MemoryStorage<TYPES, SequencingLeaf<TYPES>>,
+        ConsensusMessage = SequencingMessage<TYPES, NODE>,
+    >,
+> where
+    <TYPES as NodeType>::StateType: TestableState,
+    <TYPES as NodeType>::BlockType: TestableBlock,
+    SequencingLeaf<TYPES>: TestableLeaf,
+    HotShot<TYPES::ConsensusType, TYPES, NODE>: ViewRunner<TYPES, NODE>,
+    Self: Sync,
+{
+    /// Initializes networking, returns self
+    async fn initialize_networking(
+        config: NetworkConfig<TYPES::SignatureKey, TYPES::ElectionConfigType>,
+    ) -> Self;
+
+    /// Initializes the genesis state and HotShot instance; does not start HotShot consensus
+    /// # Panics if it cannot generate a genesis block, fails to initialize HotShot, or cannot
+    /// get the anchored view
+    /// Note: sequencing leaf does not have state, so does not return state
+    async fn initialize_state_and_hotshot(&self) -> HotShotHandle<TYPES, NODE> {
+        let genesis_block = TYPES::BlockType::genesis();
+        let initializer =
+            hotshot::HotShotInitializer::<TYPES, SequencingLeaf<TYPES>>::from_genesis(
+                genesis_block,
+            )
+            .expect("Couldn't generate genesis block");
+
+        let config = self.get_config();
+
+        let (pk, sk) =
+            TYPES::SignatureKey::generated_from_seed_indexed(config.seed, config.node_index);
+        let ek = jf_primitives::aead::KeyPair::generate(&mut rand_chacha::ChaChaRng::from_seed(
+            config.seed,
+        ));
+        let known_nodes = config.config.known_nodes.clone();
+
+        let da_network = self.get_da_network();
+        let quorum_network = self.get_quorum_network();
+
+        // Since we do not currently pass the election config type in the NetworkConfig, this will always be the default election config
+        let quorum_election_config = config.config.election_config.clone().unwrap_or_else(|| {
+            <QuorumEx<TYPES,NODE> as ConsensusExchange<
+                TYPES,
+                Message<TYPES, NODE>,
+            >>::Membership::default_election_config(config.config.total_nodes.get() as u64)
+        });
+
+        let committee_election_config = <CommitteeEx<TYPES, NODE> as ConsensusExchange<
+            TYPES,
+            Message<TYPES, NODE>,
+        >>::Membership::default_election_config(
+            config.config.total_nodes.get() as u64
+        );
+
+        let exchanges = NODE::Exchanges::create(
+            known_nodes.clone(),
+            (quorum_election_config, committee_election_config),
+            (quorum_network.clone(), da_network.clone()),
+            pk.clone(),
+            sk.clone(),
+            ek.clone(),
+        );
+        let hotshot = HotShot::init(
+            pk,
+            sk,
+            config.node_index,
+            config.config,
+            MemoryStorage::empty(),
+            exchanges,
+            initializer,
+            NoMetrics::boxed(),
```

Comment:
Do we have any plans when we benchmark to include metrics?

Reply:
Ideally we would, but realistically I don't think we'll have the time to do it.